charms/kserve-controller/src/charm.py

```diff
@@ -123,30 +123,27 @@ def __init__(self, *args):
         self._ingress_gateway_requirer = GatewayRequirer(self, relation_name="ingress-gateway")
         self._local_gateway_requirer = GatewayRequirer(self, relation_name="local-gateway")
 
-        self.framework.observe(self.on.install, self._on_install)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.on.remove, self._on_remove)
-        self.framework.observe(
-            self.on.kserve_controller_pebble_ready, self._on_kserve_controller_ready
-        )
-        self.framework.observe(
-            self.on.kube_rbac_proxy_pebble_ready, self._on_kube_rbac_proxy_ready
-        )
-        self.framework.observe(
-            self.on["ingress-gateway"].relation_changed, self._on_ingress_gateway_relation_changed
-        )
         # Observe if relation is removed by juju remove-application or juju remove-relation
         self.framework.observe(
             self.on["ingress-gateway"].relation_broken, self._on_ingress_gateway_relation_broken
         )
-        self.framework.observe(
-            self.on["local-gateway"].relation_changed, self._on_local_gateway_relation_changed
-        )
         self.framework.observe(
             self.on["local-gateway"].relation_broken, self._on_local_gateway_relation_broken
         )
+
+        for event in [
+            self.on.install,
+            self.on.config_changed,
+            self.on.kserve_controller_pebble_ready,
+            self.on.kube_rbac_proxy_pebble_ready,
+            self.on["local-gateway"].relation_changed,
+            self.on["ingress-gateway"].relation_changed,
+        ]:
+            self.framework.observe(event, self._on_event)
+
         for rel in ["object-storage", "secrets", "service-accounts"]:
-            self.framework.observe(self.on[rel].relation_changed, self._on_install)
+            self.framework.observe(self.on[rel].relation_changed, self._on_event)
```

Comment:
I vote we fold this into the `for event in []` on the line above.  Not that it is wrong to be separate, but it feels odd to have two for loops for almost the same thing

Reply:
you're right, that way is it's more readable